### ISSUE TITLE
Component System: Continue Integration

### ISSUE THREAD

**Issue by ItsJonQ** [WordPress/gutenberg#28399]
This is an extension of [Implementing a Component System](https://github.com/WordPress/gutenberg/issues/27484)

---

### Context ✨

The [`FontSizePicker`](https://github.com/WordPress/gutenberg/pull/27594) component was first to receive the upgrade to use the [new Components System](https://g2components.wordpress.com/).

We opted for a strategy (proposed by @gziolo) that imports and uses the various system parts via [npm dependencies](https://github.com/WordPress/gutenberg/issues/27484#issuecomment-738857333), rather than migrating the code by copying/pasting them over.

This was largely to expedite us figuring out the process for integration.
Using the new Component System code via external packages is not the end goal.

---

### Back Filling 🔙

@gziolo has identified the component dependencies of `FontSizePicker` in [a comment](https://github.com/WordPress/gutenberg/pull/27594#issuecomment-764504976).

We're in a place where we can work on "back filling" these components so that

* `FontSizePicker` would no longer depend on external component dependencies
* We'll establish the new foundation for the Component System

---

### Components List 📦

Below is the list of components we'll need to integrate into `@wordpress/components` from the [new component system library](https://github.com/ItsJonQ/g2/tree/master/packages/components/src).

These have been ranked in dependency order, as well as complexity:

* [x] View (#28176)
* [x] Truncate (#28176)
* [x] VisuallyHidden (#28887)
* [x] Grid(#28531)
* [ ] Icon (#28978)
* [x] Text (#28475)
* [x] Flex (#28609)
* [x] HStack (#28707)
* [x] VStack (#28798)
* [x] FormGroup (#28568)
* [x] ControlLabel (#28568)
* [x] ControlGroup (#28982)
* [x] Elevation (#28593)
* [x] Surface (#28596)
* [x] Scrollable (#28979)
* [x] Portal (#28981)
* [x] Spinner (#29146)
* [x] Card (#29084)
* [x] Popover (#29084)
* [x] Tooltip (#29385)
* [x] Heading (#29592)
* [x] ~Menu~ Replaced by `ItemGroup`
* [ ] Dropdown (BLOCKED by Card work)
* [x] Divider (#29433)
* [x] Separator
* [ ] Slider (BLOCKED by TextInput as they depend on the same useControlledValue hook)
* [x] BaseButton (#29230)
* [x] Button (#29230)
* [x] ButtonGroup (#29230)
* [x] BaseField (https://github.com/WordPress/gutenberg/pull/32250)
* [ ] TextInput (https://github.com/WordPress/gutenberg/pull/32381)
* [ ] SelectDropdown

I've added this to the running G2 Components integration spreadsheet to help track the progress:
https://docs.google.com/spreadsheets/d/1gfzznyJtTr3hHoGkGRdeBRjhuZVMRalF9NuLrxkl4Nk/edit?usp=sharing

To get a visual sense of how these components are composed together, you can check out this visual flow map:
https://miro.com/app/board/o9J_khVkF9A=/?moveToWidget=3074457351837142359&cot=14

---

### Contribution 🙏

Help in this effort is definitely welcome!

I recognize that there's a lot that goes into the new Component System.

I've recorded a walkthrough of it on the project blog:
https://g2components.wordpress.com/2020/12/02/walkthrough-integration-ideas/

If you're interested in helping and you're having trouble navigating the space, please let me know!
I'd be happy to help however I can.

I recorded a walkthrough tutorial on how we're moving these components (note: some of the things may be outdated. I'll need to re-record this, but it may help as a starting point)
https://g2components.wordpress.com/2021/01/27/moving-components-to-gutenberg-walkthrough/

**Comment by gziolo** [comment]
I'm copying over some issues I raised while working on #28624::

- [ ] `FontSizeControl` isn't typed and therefore it can't be included in the export in `packages/components/src/ui/index.js`
- [ ] `FontSizeControl` and `View` don't have stories
- [ ] `FonteSizeControl` and `View` don't have unit tests
- [ ] All React hooks included for G2 components aren't covered with unit tests
  - [ ] `useFontSizeControl`
  - [ ] `useGrid`
  - [ ] `useText`
  - [ ] `useTruncate`
  - [ ] `useView`
- [ ] It would be great to cover general-purpose utils (`createComponents`, `getOptimalTextShade`) with unit tests
- [ ] Should individual utils be covered with unit tests as well? 

I also opened #28661 that proposes the removal of the temporary feature flag that excludes G2 components codebase from the upcoming WordPress release.

**Comment by ItsJonQ** [comment]
@gziolo +100 on concerns for testing. We need to go back to add tests for these component hooks 👍 (as well as new things coming into Gutenberg)

**Comment by sarayourfriend** [comment]
Added several missing components from here that are dependencies of `FontSizeControl`, this list keeps growing 😱

**Comment by ItsJonQ** [comment]
Alrighty! We have another batch of updates ready for merge (thanks a bunch @sarayourfriend !)

* Surface (#28596)
* ControlGroup (#28982) 
* Portal (#28981)

I think they should be merged in that order.
Although they don't depend on each other, I think it'll make it easier (from a `package.json`/`package-lock.json` perspective).

**Comment by sarayourfriend** [comment]
Removing the Good First Issue label because this is no longer a good first issue as demonstrated by @ramonjd and @ntsekouras's attempts to integrate some components.

**Comment by sarayourfriend** [comment]
Closing this issue in favor of the bigger tracking issue at https://github.com/WordPress/gutenberg/issues/30503

All the remaining components from this issue have been moved over to that tracking issue under the "Finish the FontSizePicker integration" heading
